### DISCUSSION TITLE
How about a gallery plugin?

### DISCUSSION THREAD

**Discussion by i-do** [datenstrom/community#70] [Ask a question] [answered]
The best gallery I've ever seen is [fotorama.io](http://fotorama.io).
It's free and opensource and packed with cool features: touchscreen gestures support, youtube and other video files can be played as slideshow, automatic resize different graphics, fullscreen expand and so on. 

I think you can make a great universal plugin with it.

**Comment by markseuffert** [comment]
Thanks for your feedback! 

I added an imagegallery plugin. It's experimental, you can try it out and add  JavaScript/CSS code to create a gallery. Are there more open source galleries that could be worth to look at? Any favourites?

**Comment by nibreh** [comment]
Hello. There is Justified Gallery : http://miromannino.github.io/Justified-Gallery/

**Comment by dieli** [comment]
Thanks for the imagegallery plugin. I have setup my galleries like
- media
  - downloads
  - images
    - 2015_06_XXX
    - 2015_07_YYY

and I want to pass the plugin 2015_06_XXX/*.jpg. But I get all the images from both directories. Is it possible to extend it, so that I can pass a path and a regex? It seems, it currently loops over all folders and searches in each folder for the regex.

Another idea would be add options for the tag to use (currently we get ul/li, but some galleries prefer div/a tags, or other combination).

**Comment by Jan-Rueter** [comment]
Nice! But pinch zoom on iOS and it is broken, as with most plugins of this kind.

**Comment by markseuffert** [comment]
Together with help from @nogginfuel I am looking into fotorama and other image galleries again. For an image gallery with slider/carousel there's [fotorama](https://github.com/artpolikarpov/fotorama) and [flickity](https://github.com/metafizzy/flickity). For an image gallery with popup/lightbox I found [blueimp-gallery](https://github.com/blueimp/Gallery), [magnific-popup](https://github.com/dimsemenov/Magnific-Popup) and [photoswipe](https://github.com/dimsemenov/PhotoSwipe). Last one has nice touchscreen support. 

All are available via [CDN](https://cdnjs.com) and would make small plugins. Any suggestion or wishes?

**Comment by nibreh** [comment]
photoswipe is really cool ;) 

And [Fancybox](http://fancyapps.com/fancybox/) ? But maybe it's not what you are looking for ...

**Comment by nogginfuel** [comment]
I obviously have a bias toward Fotorama since I'm already using it and have
designed an initial version of the plugin. I don't know that I have a
preference on the popup/lightbox style.

Cheers,
Jef

Jef Lippiatt
UI / UX Designer
312.339.4972
nogginfuel.com
twitter: @nogginfuel http://twitter.com/nogginfuel

**Comment by markseuffert** [comment, marked as answer]
I added a [gallery extension](https://github.com/annaesvensson/yellow-gallery) which replaces the old one. It supports subdirectories like @dieli suggested, it's also touchscreen friendly. The next step is to make a fotorama plugin for a slider/carousel style.

**Comment by BsNoSi** [comment]
I there a simple way to add the „zoom“ function of the galery as a standard behaviour to _all images_ as a preset? This would enhance the usability of each page and there would be no additional coding required for images  (simply img tag as trigger).

**Comment by markseuffert** [comment]
What do you mean with "zoom"?

**Comment by BsNoSi** [comment]
Click → enlarge :: Click → resize to css-width/height.

Idea behind: I have a standard format for „normal“ images (max-width/height, right align, flow left) but often the image is larger. It would be nice, to have the same behaviour fort these images like for images in the galery.

**Comment by markseuffert** [comment]
Yes that's possible. You can use the gallery extension for single images, here's an example:

```
[gallery yellow.jpg]
```

As a developer you can also write your own extension and create exactly the HTML output you need.

**Comment by BsNoSi** [comment]
Thanks for this sample. I know this option. I want this `[gallery yellow.jpg]` as a standard behaviour for `![Alternative text](path to image)`. The _gallery tag_ is not part of markdown standard.

Currently I have not the time (and in general the experience) to create a plugin by myself. Maybe end of year I will try.
